src/hooks/shards/useShardsList.ts

```diff
@@ -40,18 +66,53 @@ const useShardsList = (catalogNames: string[]) => {
             return fetchShardList(name, session, reactorAuthorization);
         });
 
-        const shardResponses = await Promise.all(shardPromises);
+        let allCallsFailed = true;
+        const shardResponses = await Promise.allSettled(shardPromises);
+        shardResponses.forEach((shardResponse, index) => {
+            const catalogName = catalogsToFetch[index];
+
+            if (
+                isPromiseFulfilledResult(shardResponse) &&
+                shardResponse.value.shards.length > 0
+            ) {
+                allCallsFailed = false;
+                failureCountsRef.current[catalogName] &&= 0;
+                response.shards = response.shards.concat(
+                    shardResponse.value.shards
+                );
+            } else {
+                failureCountsRef.current[catalogName] =
+                    (failureCountsRef.current[catalogName] ?? 0) + 1;
 
-        const response: { shards: Shard[] } = { shards: [] };
+                const failureCount = failureCountsRef.current[catalogName];
 
-        shardResponses.forEach(({ shards }) => {
-            response.shards = response.shards.concat(shards);
+                logRocketEvent('ShardsList', {
+                    catalogName,
+                    failureCount: failureCountsRef.current[catalogName],
+                    willRetry: failureCount < MAX_FAILURES,
+                });
```

Comment:
If there was no fulfillment then keep a count of that and then fire an event so we can keep an eye on this.